testng-asserts/src/test/java/org/testng/AssertTest.java

```diff
@@ -602,6 +602,16 @@ public void testAssertEqualsDeepSetFail() {
     Assert.assertEqualsDeep(actualSet, expectedSet);
   }
 
+  @Test(description = "GITHUB-3227", expectedExceptions = AssertionError.class)
+  public void testAssertEqualsNoOrderCollection() {
+    var actual = List.of("a", "b", "b");
+    var expected = List.of("a", "b", "c");
+    Assert.assertEqualsNoOrder(actual, expected, "list should not be equal");
+    var actualQueue = new ArrayDeque<>(actual);
+    var expectedQueue = new ArrayDeque<>(expected);
+    Assert.assertEqualsNoOrder(actualQueue, expectedQueue, "queue should not be equal");
```

Comment:
This assertion is never checked because assertion error is thrown before.

Reply:
My bad. Fixed it.